File: store_server/src/main/java/org/project/store_server/service/StoreService.java
package org.project.store_server.service;

import org.project.store_server.model.dto.store.StoreRequestDto;
import org.project.store_server.model.dto.store.StoreResponseDTO;
import java.util.List;


public interface StoreService {
    List<StoreResponseDTO> getAllStores();
    StoreResponseDTO getStore(Long storeId);
    void deleteStore(Long storeId);
    void addStore(StoreRequestDto storeRequestDto);
    StoreResponseDTO updateStore(Long storeId, StoreRequestDto storeRequestDto);
    void deleteAllStores();
}

File: store-frontend/src/app/services/store.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Store, StoreRequest } from '../models/store.model';
import { Stock } from '../models/stock.model';

@Injectable({
  providedIn: 'root'
})
export class StoreService {
  private apiUrl = 'http://localhost:8080/stores';

  constructor(private http: HttpClient) { }

  getAllStores(): Observable<Store[]> {
    return this.http.get<Store[]>(this.apiUrl);
  }

  getStore(id: number): Observable<Store> {
    return this.http.get<Store>(`${this.apiUrl}/${id}`);
  }

  createStore(store: StoreRequest): Observable<void> {
    return this.http.post<void>(this.apiUrl, store);
  }

  updateStore(id: number, store: StoreRequest): Observable<Store> {
    return this.http.put<Store>(`${this.apiUrl}/${id}`, store);
  }

  deleteStore(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }

  deleteAllStores(): Observable<void> {
    return this.http.delete<void>(this.apiUrl);
  }

  getStoreStocks(storeId: number): Observable<Stock[]> {
    return this.http.get<Stock[]>(`${this.apiUrl}/${storeId}/stocks`);
  }
}